drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationGridHeaderUtilities.java

```diff
@@ -151,10 +152,18 @@ public static String getExistingInstances(final String group, final ScenarioGrid
                 .collect(Collectors.toSet()));
     }
 
-    public static List<String> getPropertyNameElements(final Simulation simulation, final int columnIndex) {
-        return Collections.unmodifiableList(simulation.getSimulationDescriptor().getFactMappingByIndex(columnIndex).getExpressionElementsWithoutClass()
-                                                    .stream()
-                                                    .map(ExpressionElement::getStep)
-                                                    .collect(Collectors.toList()));
+    public static List<String> getPropertyNameElements(final ScenarioGridModel scenarioGridModel, final int columnIndex) {
+        final Optional<Simulation> optionalSimulation = scenarioGridModel.getSimulation();
+        return optionalSimulation.map(simulation -> {
+            final FactMapping factMapping = scenarioGridModel.getSimulation().get().getSimulationDescriptor().getFactMappingByIndex(columnIndex);
```

Comment:
You can use `simulation` here instead of access again via `scenarioGridModel.getSimulation()...`

Reply:
@danielezonca Done, thank you!